climpred_skeleton/time.py

```diff
@@ -0,0 +1,89 @@
+import warnings
+from typing import Union
+
+import cftime
+import pandas as pd
+import xarray as xr
+
+from climpred_skeleton.core import Verification
+
+
+class TimeManager(Verification):
+    def __init__(self, initialized, observation):
+        super().__init__(initialized, observation)
+        # Convert `init` and `time` indices to CFTimeIndex.
+        self._convert_to_cftime_index()
+        self._units = self._initialized['lead'].attrs['units']
+
+    def _convert_to_cftime_index(self, calendar: str = 'DatetimeProlepticGregorian'):
+        """Converts time indices for the prediction and observations to a
+        CFTimeIndex."""
+
+        def _return_converted_time_index(
```

Comment:
I personally don't like nested functions besides for decorators and I haven't seen them use in other repos I've worked on, although I guess that's fairly limited.

If there's a need for nested functions, maybe it's a sign to make a class off of this main class, similar to the link you shared with serializing different json/xml/csv formats

Reply:
I always don’t mind nested functions. What else to do? A function structured code. And nested because nobody else needs this

Reply:
I side with Aaron on this one. I think it's a nice way to keep code a little cleaner